domain/ide/workspace/folder.go

```diff
@@ -479,46 +479,42 @@ func appendTestResults(sic snyk.SeverityIssueCounts, results []json_schemas.Test
 	return results
 }
 
-func (f *Folder) FilterAndPublishDiagnostics(p *product.Product) {
-	productIssuesByFile, err := f.getDelta(f.IssuesByProduct(), p)
+func (f *Folder) FilterAndPublishDiagnostics(p product.Product) {
+	issueByProduct := f.IssuesByProduct()
+	if len(issueByProduct[p]) == 0 {
+		return
+	}
```

Comment:
we need to send publishDiagnostics with an empty diagnostics array for files that don't have diagnostics anymore.